tests/test_switch.py

```diff
@@ -0,0 +1,18 @@
+from textual.app import App, ComposeResult
+from textual.widgets import Switch
+
+
+async def test_switch_click_doesnt_bubble_up():
+    """Regression test for https://github.com/Textualize/textual/issues/2366"""
+
+    class SwitchApp(App[None]):
+        def compose(self) -> ComposeResult:
+            yield Switch()
+
+        async def on_click(self) -> None:
+            raise AssertionError(
```

Comment:
Can you check this fails without the fix? I seem to recall an issue where assertion errors where swallowed when raised via pilot. (just being paranoid)

Reply:
Ahhh I see. I seem to recall Darren taking care of this, but you can never be too sure.
@seifertm, let us know what you find.

Reply:
I'm equally paranoid about these things, so I wanted to see my test fail without the fix as well. I commented out `event.stop()` and got this stack trace for the test when running `pytest -k test_switch`:

```
self = SwitchApp(title='SwitchApp', classes={'-dark-mode'})

    async def on_click(self) -> None:
>       raise AssertionError(
            "The app should never receive a click event when Switch is clicked."
        )
E       AssertionError: The app should never receive a click event when Switch is clicked.

tests/test_switch.py:13: AssertionError
-----Captured stderr call -----
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /home/michael/oss/textual/tests/test_switch.py:13 in on_click                │
│                                                                              │
│   10 │   │   │   yield Switch()                                              │
│   11 │   │                                                                   │
│   12 │   │   async def on_click(self) -> None:                               │
│ ❱ 13 │   │   │   raise AssertionError(                                       │
│   14 │   │   │   │   "The app should never receive a click event when Switch │
│   15 │   │   │   )                                                           │
│   16                                                                         │
│                                                                              │
│ ╭────────────────────────── locals ───────────────────────────╮              │
│ │ self = SwitchApp(title='SwitchApp', classes={'-dark-mode'}) │              │
│ ╰─────────────────────────────────────────────────────────────╯              │
╰──────────────────────────────────────────────────────────────────────────────╯
AssertionError: The app should never receive a click event when Switch is 
clicked.
----- snapshot report summary -----
1 snapshot passed.
===== short test summary info =====
FAILED tests/test_switch.py::test_switch_click_doesnt_bubble_up - AssertionError: The app should never receive a click event when Switch is clicked.
====== 1 failed, 6 passed, 1518 deselected in 2.42s =====
(
```

Reply:
Works for me!